tests/experimental/datasources/test_postgres_datasource.py

```diff
@@ -493,28 +495,45 @@ def test_sort_batch_list_by_unknown_key(create_source):
             source.get_batch_list_from_batch_request(batch_request)
 
 
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "order_by",
+    [
+        ["+year", "-month"],
+        [{"metadata_key": "year"}, {"metadata_key": "month", "reverse": True}],
+    ],
+)
+def test_table_asset_sorter_parsing(order_by: list):
+    """Ensure that arguments to `order_by` are parsed correctly regardless if they are lists of dicts or a list of strings"""
+    expected_sorters = [
+        BatchSorter(metadata_key="year"),
+        BatchSorter(metadata_key="month", reverse=True),
+    ]
+
+    table_asset = TableAsset(
+        name="SorterTest", table_name="SORTER_TEST", order_by=order_by
+    )
+    print(table_asset)
+    pprint(f"\n{table_asset.dict()}")
+
+    assert table_asset.order_by == expected_sorters
+
+
 @pytest.mark.unit
 def test_data_source_json_has_properties(create_source):
     with create_source(lambda _: None) as source:
-        assert isinstance(TableAsset.order_by, property), (
-            "This test assumes TableAsset.order_by is a property. If it is not we "
-            "should update this test",
-        )
         asset = source.add_table_asset(name="my_asset", table_name="my_table")
         asset.add_year_and_month_splitter(column_name="my_col").add_sorters(
             ["year", "month"]
         )
-        source_json = source.json()
+        source_json = source.json(indent=4, sort_keys=True)
+        print(source_json)
```

Comment:
Should this be `pprint`?

Reply:
No because it's just a string. The `indent` keyword makes it pretty.